File: freetools/tools/ocr.py
import pytesseract
from PIL import Image
import io
import os

# Jika di Windows, tentukan path Tesseract secara manual
if os.name == "nt":
    pytesseract.pytesseract.tesseract_cmd = r"C:\Program Files\Tesseract-OCR\tesseract.exe"

def extract_text(image_bytes):
    image = Image.open(io.BytesIO(image_bytes))
    text = pytesseract.image_to_string(image)
    return text if text.strip() else "Teks tidak ditemukan atau tidak terbaca."

print(f"OCR bisa")

File: freetools/tools/ocr.js
const express = require("express");
const multer = require("multer");
const Tesseract = require("tesseract.js");
const cors = require("cors");

const app = express();
app.use(cors());

const upload = multer({ storage: multer.memoryStorage() });

app.post("/ocr", upload.single("image"), (req, res) => {
    if (!req.file) return res.status(400).json({ error: "File tidak ditemukan." });

    Tesseract.recognize(req.file.buffer, "eng", {
        logger: (m) => console.log(m),
    }).then(({ data: { text } }) => {
        res.json({ text: text.replace(/\n/g, "<br>") || "Teks tidak ditemukan atau tidak terbaca." });
    }).catch(err => res.status(500).json({ error: err.toString() }));
});

const PORT = 5001;
app.listen(PORT, () => {
    console.log(`OCR Server berjalan di http://localhost:${PORT}`);
});
